DOCS-1578: Add data management and capture steps to Cartographer SLAM page (#2396)
viamrobotics/docs: docs/mobility/slam/cartographer/_index.md

Reviewer: Small optional nit, really more of a question - do you want to include a link to the [Viam Rover](https://www.viam.com/resources/rover) on line 58?

Line 58:
>  For example, if you are using a Viam Rover and the [RPlidar A1](https://www.slamtec.com/en/Lidar/A1) model, mount it to the Rover so that the **pointed** end of the RPlidar mount housing is facing in the same direction as the webcam.

Suggestion:
> For example, if you are using a [Viam Rover](https://www.viam.com/resources/rover) and the [RPlidar A1](https://www.slamtec.com/en/Lidar/A1) model, mount it to the Rover so that the **pointed** end of the RPlidar mount housing is facing in the same direction as the webcam.

Author: Done! good idea

Reviewer: Tiny nit on word order, really subjective! I thought it might be nice if they read "only" before the end of the sentence.

Line 72:
>   Currently, the `rplidar` and `cartographer` modules support the Linux platform only.

Suggestion: 
>   Currently, the `rplidar` and `cartographer` modules  only support the Linux platform.

Author: Done

Author: good w me~~

Author: done!

   2. Click **Create service** in the lower-left corner of the page.
   Choose `Data Management` as the type and specify a name for your data management service, for example `Data-Management-Service`.
   3. Click **Create**.
   4. On the panel that appears, you can manage the capturing and syncing functions and specify the **directory**, the sync **interval** and any **tags** to apply to captured data. See the [data management service](/data/) for more information.
  
   Enable data capture for your camera and/or movement sensor:

   5. Find the component's card on your machine's **Config** tab.
   6. Click `Add Method` and then select the method type and the capture frequency. For an RPlidar, choose the method [`NextPointCloud`](http://localhost:1313/components/camera/#getpointcloud).
   We recommend a capture frequency of `5` Hz for RPlidar cameras and `20` Hz for movement sensors.

   Set up the `cartographer` module on your machine:

   7. Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com).
   8. Click on the **Services** subtab and click **Create service** in the lower-left corner.

>    9.  Select **SLAM**, then select `cartographer`.

Author: ```suggestion
   9. Select **SLAM**, then select `cartographer`.
```

To create a new map, follow the instructions below.
Creating a new map uses an instance of the cartographer module running in the cloud.

1. Enable data capture and configure your `cartographer` SLAM service

   After installing your physical RPlidar and adding the `rplidar` module as outlined in the [requirements](#requirements) section, follow the steps below to enable data capture and add the `cartographer` module to your machine:

   {{< tabs name="Create new map">}}
   {{% tab name="Config Builder" %}}

   Add the data management service:

   1. From your machine's **Config** tab, navigate to the **Services** subtab.
   2. Click **Create service** in the lower-left corner of the page.
      Choose `Data Management` as the type and specify a name for your data management service, for example `Data-Management-Service`.
   3. Click **Create**.
   4. On the panel that appears, you can manage the capturing and syncing functions and specify the **directory**, the sync **interval** and any **tags** to apply to captured data. See the [data management service](/data/) for more information.

>    Enable data capture for your camera and/or movement sensor:

Author: Adding

   5. Find the component's card on your machine's **Config** tab.

>    6. Click `Add Method` and then select the method type and the capture frequency. For an RPlidar, choose the method [`NextPointCloud`](/components/camera/#getpointcloud).

Author: Adding, thanks!

>    6. Click `Add Method` and then select the method type and the capture frequency. For an RPlidar, choose the method [`NextPointCloud`](http://localhost:1313/components/camera/#getpointcloud).

Author: ```suggestion
   6. Click `Add Method` and then select the method type and the capture frequency. For an RPlidar, choose the method [`NextPointCloud`](/components/camera/#getpointcloud).
```

      We recommend a capture frequency of `5` Hz for RPlidar cameras and `20` Hz for movement sensors.

> 

Reviewer 2: What do you think of moving the alert that says:

```
Be aware that data is not only captured when a slam session is running.
[Data Capture](/data/capture/) continuously monitors and captures your machine's sensor data while the machine is running.
To avoid incurring charges while not in use, [turn off data capture for your sensors](/data/capture/).
```

up to here, next to where we now talk about adding data capture?

Author: Sounds good, moved

To create a new map, follow the instructions below.
Creating a new map uses an instance of the cartographer module running in the cloud.

> 1. Enable data capture and configure your `cartographer` SLAM service

Reviewer 3: If captured here we need to also capture it on the matching json config tab

To create a new map, follow the instructions below.
Creating a new map uses an instance of the cartographer module running in the cloud.

1. Enable data capture and configure your `cartographer` SLAM service

   Follow the steps below to enable data capture and add the `cartographer` module to your machine:

   {{< tabs name="Create new map">}}
   {{% tab name="Config Builder" %}}

   Add the data management service:

>    1. From your machine's **Config** tab, navigate to the **Services** subtab.

Reviewer: Small optional nit for consistency in this list..! 
```suggestion
   1. Navigate to the **Services** subtab on your machine's **Config** tab.
```

   2. Click **Create service** in the lower-left corner of the page.
      Choose `Data Management` as the type and specify a name for your data management service, for example `Data-Management-Service`.
   3. Click **Create**.

>    4. On the panel that appears, you can manage the capturing and syncing functions and specify the **directory**, the sync **interval** and any **tags** to apply to captured data. See the [data management service](/data/) for more information.

Reviewer: I really love the detail in this. 

I thought possibly dividing it into two sentence might make it a bit easier to digest... what do you think though? Does this still retain your original meaning? Either way very subjective and optional. 

```suggestion
   4. On the panel that appears, you can manage the capturing and syncing functions. You can specify the **directory**, sync **interval**, and any **tags** to apply to captured data. See the [data management service](/data/) for more information.
```

   Enable data capture for your camera, and for your movement sensor if you would like to use a movement sensor to provide IMU input:

   5. Find the component's card on your machine's **Config** tab.

>    6. Click `Add Method` and then select the method type and the capture frequency. For the camera, choose the [`NextPointCloud`](/components/camera/#getpointcloud) method. For a movement sensor, choose the [`AngularVelocity`](/components/movement-sensor/#getangularvelocity) and [`LinearAcceleration`](/components/movement-sensor/#getlinearacceleration) methods.

Reviewer: Small optional nit! I thought changing "select" to "specify" would make the verb applicable to both "method type" and "capture frequency". 
```suggestion
   6. Click `Add Method`, then specify the method type and the capture frequency. For the camera, choose the [`NextPointCloud`](/components/camera/#getpointcloud) method. For a movement sensor, choose the [`AngularVelocity`](/components/movement-sensor/#getangularvelocity) and [`LinearAcceleration`](/components/movement-sensor/#getlinearacceleration) methods.
```

Author: Done!

      We recommend a capture frequency of `5` Hz for RPlidar cameras and `20` Hz for movement sensors.
   

>    {{< alert title="Tip" color="tip" >}}

Reviewer 2: Thanks for moving this up here! This alert also currently appears under "Update an existing map", but I think we could remove it from there and just have it once here

Author: I also found an alert in the "Localize Only" section and removed, is that ok or would you want to keep that one in? @EshaMaharishi

Reviewer 2: Oh, good catch, not sure why that alert was there under "Localize Only" since data capture isn't even used there. Sounds good to remove!

>    Be aware that data is not only captured when a slam session is running.

Reviewer: Super optional small nit. Do you think adding "data capture" could make it more clear which feature this is referencing/where in the app they would find it since it matches the name they'd see on the app?  Could also not be needed since the next sentence mentions it so explicitly!

```suggestion
   Be aware that data capture is not restricted to when a slam session is running.
```

Author: Not implementing this just bc esha suggestion overwrites this sentence

   [Data Capture](/data/capture/) continuously monitors and captures your machine's sensor data while the machine is running.
   To avoid incurring charges while not in use, [turn off data capture for your sensors](/data/capture/).
   {{< /alert >}}

   Set up the `cartographer` module on your machine:

   7. Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com).
   8. Click on the **Services** subtab and click **Create service** in the lower-left corner.
   9. Select **SLAM**, then select `cartographer`.
      You can also search for "cartographer".
   10. Click **Add module**, give your service a name of your choice, then click **Create**.

>    11. In the resulting `SLAM` service configuration pane, first choose `Create new map` as the **Mapping mode**, then configure the rest of the **Attributes** for that mapping mode:

Reviewer: small nit
```suggestion
   11. In the resulting `SLAM` service configuration pane, choose `Create new map` as the **Mapping mode**, then configure the rest of the **Attributes** for that mapping mode:
```

Author: nice

   - adds the `viam:rplidar` and the `viam:cartographer` modules
   - configures the `viam:slam:cartographer` service and the [data management service](/data/)

>    - adds an `viam:lidar:rplidar` camera with data capture configured

Reviewer 2: Should we make this same change for "Update existing map" below?

    {{< tabs name="Localize only">}}
    {{% tab name="Config Builder" %}}

The configuration is similar to the configuration for [updating an existing map](#update-an-existing-map), except instead of adding a data management service and configuring data capture on the camera and movement sensor, set a `Data polling rate (Hz)` on both.

> The `cartographer` module on your machine polls the live LiDAR and IMU directly at these rates, whereas the capture rate is only used when data is being sent to the cloud.

Reviewer 2: Suggestion: "the capture rate" -> "data capture"

Author: Changed

> The `cartographer` module on your machine polls the live LiDAR and IMU directly at these rates, whereas the capture rate is only used when data is being sent to the cloud.

Author: ```suggestion
The `cartographer` module on your machine polls the live LiDAR and IMU directly at these rates, whereas data capture is only used when data is being sent to the cloud.
```
